internal/store/serverless_instances.go

```diff
@@ -19,42 +19,60 @@ import (
 
 	"github.com/mongodb/mongodb-atlas-cli/internal/config"
 	atlas "go.mongodb.org/atlas/mongodbatlas"
+	atlasv2 "go.mongodb.org/atlas/mongodbatlasv2"
 )
 
 //go:generate mockgen -destination=../mocks/mock_serverless_instances.go -package=mocks github.com/mongodb/mongodb-atlas-cli/internal/store ServerlessInstanceLister,ServerlessInstanceDescriber,ServerlessInstanceDeleter,ServerlessInstanceCreator,ServerlessInstanceUpdater
 
 type ServerlessInstanceLister interface {
-	ServerlessInstances(string, *atlas.ListOptions) (*atlas.ClustersResponse, error)
+	ServerlessInstances(string, *atlas.ListOptions) ([]atlasv2.ServerlessInstanceDescription, error)
 }
 
 type ServerlessInstanceDescriber interface {
 	ServerlessInstance(string, string) (*atlas.Cluster, error)
+	GetServerlessInstance(string, string) (*atlasv2.ServerlessInstanceDescription, error)
 }
 
 type ServerlessInstanceDeleter interface {
 	DeleteServerlessInstance(string, string) error
 }
 
 type ServerlessInstanceCreator interface {
-	CreateServerlessInstance(string, *atlas.ServerlessCreateRequestParams) (*atlas.Cluster, error)
+	CreateServerlessInstance(string, *atlasv2.ServerlessInstanceDescriptionCreate) (*atlasv2.ServerlessInstanceDescription, error)
 }
 
 type ServerlessInstanceUpdater interface {
-	UpdateServerlessInstance(string, string, *atlas.ServerlessUpdateRequestParams) (*atlas.Cluster, error)
+	UpdateServerlessInstance(string, string, *atlasv2.ServerlessInstanceDescriptionUpdate) (*atlasv2.ServerlessInstanceDescription, error)
 }
 
 // ServerlessInstances encapsulates the logic to manage different cloud providers.
-func (s *Store) ServerlessInstances(projectID string, listOps *atlas.ListOptions) (*atlas.ClustersResponse, error) {
+func (s *Store) ServerlessInstances(projectID string, listOps *atlas.ListOptions) ([]atlasv2.ServerlessInstanceDescription, error) {
 	switch s.service {
 	case config.CloudService:
-		result, _, err := s.client.(*atlas.Client).ServerlessInstances.List(s.ctx, projectID, listOps)
-		return result, err
+		result, _, err := s.clientv2.ServerlessInstancesApi.ListServerlessInstances(s.ctx, projectID).
+			IncludeCount(listOps.IncludeCount).
+			ItemsPerPage(int32(listOps.ItemsPerPage)).
+			PageNum(int32(listOps.PageNum)).
+			Execute()
+
+		return result.Results, err
```

Comment:
you are loosing the envelop object here which means I'm pretty sure this doesn't work anymore as the template is not updated, please don't unwrap the envelop object as people use it to get the pagination information

Reply:
Oopsie - I can revert this. Just a [q] on this, is there any particular reason we are keeping this enveloped and other list responses not?

Reply:
Well paint me curious because we have been always trying to keep the envelope, only exception there were some really old (before APIx times) sdk methods that were removing it and since the sdk wasn't returning it there was little we could do.

We keep the envelope as it includes the number of pages information which otherwise would be lost

Reply:
I've seen this around, and in some of the commands we've migrated we remove the envelope from the v2 client. Should we consider returning the enveloped version of all paginated views now that these are present in the new sdk?

Reply:
>  and in some of the commands we've migrated we remove the envelope

I did a quick search of the store and no store is returning .Results like this is (a tleat on master)

>. Should we consider returning the enveloped version of all paginated views now that these are present in the new sdk?

Yes when possible but at least we should maintain behavior, which this change was breaking